.github/workflows/ci.yml

```diff
@@ -0,0 +1,226 @@
+        uses: "actions/checkout@v2"
+      - name: "Setup environment"
+        uses: "networktocode/gh-action-setup-poetry-environment@v1"
+      - name: "Install hadolint"
+        run: "curl -Lo /usr/local/bin/hadolint https://github.com/hadolint/hadolint/releases/download/v2.0.0/hadolint-Linux-x86_64 && chmod +x /usr/local/bin/hadolint"
+      - name: "Linting: hadolint"
+        run: "poetry run invoke hadolint"
+  check-migrations:
+    runs-on: "ubuntu-20.04"
+    env:
+      INVOKE_NAUTOBOT_LOCAL: "True"
+      NAUTOBOT_DB_USER: "nautobot"
+      NAUTOBOT_DB_PASSWORD: "decinablesprewad"
+      NAUTOBOT_DB_ENGINE: "django.db.backends.postgresql"
+    services:
+      postgres:
+        image: "postgres:13"
+        env:
+          POSTGRES_USER: "nautobot"
+          POSTGRES_PASSWORD: "decinablesprewad"
+          POSTGRES_DB: "nautobot"
+        ports:
+          - "5432:5432"
+    steps:
+      - name: "Check out repository code"
+        uses: "actions/checkout@v2"
+      - name: "Setup environment"
+        uses: "networktocode/gh-action-setup-poetry-environment@v1"
+      - name: "Run check migrations"
+        run: "poetry run invoke check-migrations"
+    needs:
+      - "black"
+      - "flake8"
+      - "hadolint"
+  tests-postgres:
+    runs-on: "ubuntu-20.04"
+    strategy:
+      fail-fast: true
+      matrix:
+        python-version: [ "3.6", "3.7", "3.8", "3.9" ]
+    env:
+      INVOKE_NAUTOBOT_LOCAL: "True"
+      NAUTOBOT_DB_USER: "nautobot"
+      NAUTOBOT_DB_PASSWORD: "decinablesprewad"
+      NAUTOBOT_DB_ENGINE: "django.db.backends.postgresql"
+    services:
+      postgres:
+        image: "postgres:13"
+        env:
+          POSTGRES_USER: "nautobot"
+          POSTGRES_PASSWORD: "decinablesprewad"
+          POSTGRES_DB: "nautobot"
+        ports:
+          - "5432:5432"
+      redis:
+        image: "redis:6-alpine"
+        ports:
+          - "6379:6379"
+    steps:
+      - name: "Check out repository code"
+        uses: "actions/checkout@v2"
+      - name: "Set up Python"
+        uses: "actions/setup-python@v2"
+        id: "python-setup"
+        with:
+          python-version: "${{ matrix.python-version }}"
+      - name: "Install Poetry"
+        run: "pip install poetry"
+      - name: "Load Environment From Cache"
+        uses: "actions/cache@v2"
+        id: "cached-poetry"
+        with:
+          path: "~/.cache/pypoetry/virtualenvs/"
+          key: "${{ runner.os }}-poetry-${{ steps.python-setup.outputs.python-version }}-${{ hashFiles('./poetry.lock') }}"
+      - name: "Install Dependencies"
+        run: "poetry install --no-interaction --no-root"
+      - name: "Install Nautobot"
+        run: "poetry install --no-interaction"
+      - name: "Run unittest"
+        run: "poetry run invoke unittest --failfast --keepdb"
+      - name: "Run coverage"
+        run: "poetry run invoke unittest-coverage"
+    needs:
+      - "check-migrations"
+  tests-mysql:
```

Comment:
Can we combine the tests-postgres and MySQL?  add MySQL as a service then set the env in the test container to whichever DB?

Reply:
Seems like the downside would be that we'd need to start and run both Postgres and MySQL containers in the combined matrix?

Reply:
I think that merging both jobs would be quite hard because we need to use different env vars for each database type. I mean it is doable, but this means some "if" statements. I would keep it separate. Also easier to see where the issue is.

Reply:
agreed, keep them separate for now